subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -205,11 +204,46 @@ static u32_t filter_remove(struct ll_filter *filter, u8_t addr_type,
 }
 #endif
 
-struct ll_filter *ctrl_filter_get(void)
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+u8_t *ctrl_irks_get(u8_t *count)
+{
+	*count = peer_irk_count;
+	return (u8_t *)peer_irks;
+}
+
+bool ctrl_irk_whitelisted(u8_t irkmatch_id)
+{
+	u8_t i = peer_irk_rl_ids[irkmatch_id];
+
+	LL_ASSERT(irkmatch_id < peer_irk_count);
+	LL_ASSERT(rl[i].taken);
+	return rl[i].wl;
+}
+
+bool ctrl_rl_idx_match(u8_t irkmatch_id, u8_t rl_idx)
 {
-	return &wl;
+	u8_t i = peer_irk_rl_ids[irkmatch_id];
+
+	LL_ASSERT(irkmatch_id < peer_irk_count);
+	LL_ASSERT(rl[i].taken);
+	return i == rl_idx;
+}
+#endif
+
+struct ll_filter *ctrl_filter_get(bool whitelist)
+{
+	if (whitelist) {
+		return &wl;
+	} else {
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+		return &rl_filter;
+#else
+		LL_ASSERT(0);
+#endif
+	}
```

Comment:
Same thing here regarding the else { branch, however I think checkpatch also complained of this